.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,29 @@
+name: Pre-commit Checks
+
+on: [pull_request]
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.7
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pre-commit
+
+      - name: Run pre-commit hooks in `tuning`
+        working-directory: tuning
+        run: |
+          pre-commit run --all-files
```

Comment:
Did you consider using https://github.com/pre-commit/action, instead of this explicit coding? The options used are a bit different, see https://github.com/pre-commit/action/blob/main/action.yml:
* `pre-commit run --all-files`
* `pre-commit run --show-diff-on-failure --color=always ${{ inputs.extra_args }}`

I think you should still be able to use the `working-directory` with that action